diff --git a/server/src/main/java/com/genymobile/scrcpy/FakeContext.java b/server/src/main/java/com/genymobile/scrcpy/FakeContext.java
index 7c0f364..970d7f5 100644
--- a/server/src/main/java/com/genymobile/scrcpy/FakeContext.java
+++ b/server/src/main/java/com/genymobile/scrcpy/FakeContext.java
@@ -13,0 +14 @@ import android.os.Process;
+import android.os.UserHandle;
@@ -14,0 +16 @@ import android.os.Process;
+import java.lang.reflect.Constructor;
@@ -15,0 +18 @@ import java.lang.reflect.Field;
+import java.lang.reflect.Method;
@@ -59,0 +63,17 @@ public final class FakeContext extends ContextWrapper {
+    private static Context createContextForCurrentUser() {
+        Context context = Workarounds.getSystemContext();
+        try {
+            Method getCurrentUser = Class.forName("android.app.ActivityManager").getDeclaredMethod("getCurrentUser");
+            int userId = (int) getCurrentUser.invoke(null);
+
+            Constructor<UserHandle> constructor = UserHandle.class.getDeclaredConstructor(int.class);
+            constructor.setAccessible(true);
+            UserHandle user = constructor.newInstance(userId);
+
+            Method createContextAsUser = Context.class.getDeclaredMethod("createContextAsUser", UserHandle.class, int.class);
+            return (Context) createContextAsUser.invoke(context, user, 0);
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
@@ -61 +81 @@ public final class FakeContext extends ContextWrapper {
-        super(Workarounds.getSystemContext());
+        super(createContextForCurrentUser());
